From f4cd5f7aace8ab9044a94661771e5301c8f945b3 Mon Sep 17 00:00:00 2001
From: Vicentiu Galanopulo <vicentiu@balena.io>
Date: Tue, 17 Aug 2021 16:04:56 +0200
Subject: [PATCH] Proper assignament of environment variables and bootcmd fix

With the old implementation, when the ALT_BOOT switch is in ON position,
the usb_boot command would set rootdev pointing to /dev/sda2, which resulted
in having a rootfs from the first dectected USB port of the EtcherPro, if that
USB port had an image of the BalenaOS.

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Vicentiu Galanopulo  <vicentiu@balena.io>
---
 include/configs/cl-som-imx8.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/configs/cl-som-imx8.h b/include/configs/cl-som-imx8.h
index 8fa6239..64b1867 100644
--- a/include/configs/cl-som-imx8.h
+++ b/include/configs/cl-som-imx8.h
@@ -129,6 +129,7 @@
 	"loadbootscript=load ${iface} ${dev}:${part} ${loadaddr} ${script}\0"	\
 	"loadfdt=load ${iface} ${dev}:${part} ${fdt_addr} ${fdt_file}\0"	\
 	"loadimage=load ${iface} ${dev}:${part} ${zip_addr} ${image}; unzip ${zip_addr} ${loadaddr};\0"	\
+	"mmcroot=/dev/mmcblk${mmcdev}p2 rootwait rw;\0"	\
 	"mmc_boot=setenv iface mmc; run ${iface}_pre; run ${iface}_init; run iface_args; run iface_boot\0"	\
 	"mmc_init=mmc rescan\0"	\
 	"mmc_pre=setenv iface mmc; setenv dev ${mmcdev}; setenv rootdev /dev/mmcblk${mmcdev}p2\0"	\
@@ -137,12 +138,13 @@
 	"usb_pre=setenv iface usb; setenv dev 0; setenv rootdev /dev/sda2\0"	\
 
 #define CONFIG_BOOTCOMMAND \
+	"setenv resin_uboot_devices ${mmcdev};" \
 	"setenv resin_kernel_load_addr ${loadaddr};" \
 	"run resin_set_kernel_root; run set_os_cmdline; " \
 	"setenv mmcdev ${resin_dev_index};" \
 	"setenv mmcbootpart ${resin_boot_part};" \
 	"run set_fdt_file;" \
-	"setenv mmcdev 1; run mmc_boot; run usb_boot; setenv mmcdev 0; run mmc_boot;"
+	"setenv mmcdev 1; run mmc_boot; setenv mmcdev 0; run mmc_boot;"
 
 /* Link Definitions */
 #define CONFIG_LOADADDR			0x40480000
-- 
2.7.4

